diagnostics/seaice_suite/seaice_suite_sic_mean_sigma.py

```diff
@@ -0,0 +1,415 @@
+    .. math::
+        a = residuals
+        b = residuals shifted by lag 
+        r_{ab} = \\frac{ \\sum_{i=i}^{n} (a_{i} - \\bar{a}) (b_{i} - \\bar{b}) }
+                 {\\sqrt{ \\sum_{i=1}^{n} (a_{i} - \\bar{a})^{2} }
+                  \\sqrt{ \\sum_{i=1}^{n} (b_{i} - \\bar{b})^{2} }}
+    Parameters
+    ----------
+    residual : xarray.DataArray detrending by month, which also deseasonalizes
+
+    Returns
+    -------
+    rlag : xarray.DataArray    Pearson's correlation coefficient at lag by month
+    """
+
+    rlag = xr.apply_ufunc(
+        _lagcorr,
+        residuals,
+        input_core_dims=[['year','month']],
+        kwargs={'lag': lag},
+        output_core_dims=[['month']],
+        output_dtypes=[np.float],
+        vectorize=True)
+        #dask='parallelized')
+
+    rlag.name = 'lagcorr'
+    rlag=rlag.transpose('month',...)
+        
+    return rlag
+
+
+# In[8]:
+
+
+def processandsave(field,file_out,firstyr=1979,lastyr=2014):
+    ### 2) Doing computations on model or obs: #####################################
+    # this takes about a few min on data that is already limited to the Arctic
+    # recommend using DASK if you decide to compute global metrics
+    # residuals are detrended field for each month, effectively making them detrended and deseasonalized
+    themean, thestd, trend, detrendedstd, residuals = mainmonthlystats(field,firstyr,lastyr)
+    print('main stats done')
+    onemolagcorr=lagcorr(residuals,1) 
+    oneyrlagcorr=lagcorr(residuals,12)
+    onemolagcorr.name='onemolagcorr'
+    oneyrlagcorr.name='oneyrlagcorr'
+
+    ### 3) Saving output data: #####################################################
+
+    allstats=xr.merge([themean, thestd, trend, detrendedstd, onemolagcorr, oneyrlagcorr])
+
+    months = ['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec']
+
+    allstats.coords['monthabbrev'] = xr.DataArray(months, dims='month')
+    allstats
+
+    allstats.to_netcdf(file_out)
+
+
+# In[11]:
+
+
+if processmod:
+    processandsave(field,proc_mod_file,firstyr,lastyr)
+
+if processobs:
+    processandsave(obs,proc_obs_file,firstyr,lastyr)
+
+
+# In[12]:
+
+
+### 4) Read processed data, regrid model to obs grid, plot, saving figures: #######################################
+
+obsstats = xr.open_dataset(proc_obs_file)
+obsstats=obsstats.rename({'latitude':'lat'})
+obsstats=obsstats.rename({'longitude':'lon'})
+
+obsmean=obsstats.themean 
+obsstd=obsstats.thestd
+obstrend=obsstats.trend 
+obsdetrendedstd=obsstats.detrended_std 
+obsonemolagcorr=obsstats.onemolagcorr
+obsoneyrlagcorr=obsstats.oneyrlagcorr
+
+modstats = xr.open_dataset(proc_mod_file)
+
+coords = [a for a in modstats.coords]
+if 'longitude' in coords:
+    modstats=modstats.rename({'latitude':'lat'})
+    modstats=modstats.rename({'longitude':'lon'})
+
+# regrid model data to obs grid
+method = 'nearest_s2d'       #method = 'nearest_d2s'  # this was bad do not use
```

Comment:
It looks like the commented line was a debugging test, so you can delete it if this is the case.